lib/portage/dispatch_conf.py

```diff
@@ -347,7 +336,7 @@ def file_archive(archive, curconf, newconf, mrgconf):
     # Archive the current config file if it isn't already saved
     if (
         os.path.lexists(archive)
-        and len(diffstatusoutput_mixed("diff -aq '%s' '%s'", curconf, archive)[1]) != 0
+        and diffstatusoutput_mixed(f"diff -aq '{curconf}' '{archive}'")[1]
```

Comment:
Dropping the `len(...) != 0` might be a fine change on its own, but it shouldn't be done in this patch.